pallets/api/src/fungibles/mod.rs

```diff
@@ -344,7 +345,10 @@ pub mod pallet {
 				spender_source.clone(),
 			)
 			.map_err(|e| e.with_weight(WeightOf::<T>::approve(0, 1)))?;
-			let new_allowance = current_allowance.saturating_sub(value);
+			let new_allowance = match current_allowance.checked_sub(&value) {
```

Comment:
This should error before the `cancel_approval` above so that we are not calling it for no reason.